include/zephyr/senss/senss_datatypes.h

```diff
@@ -0,0 +1,259 @@
+ *	data[0]        int32    micro g     -6              +/-4*10^6         x axis acceleration
+ *	data[1]        int32    micro g     -6              +/-4*10^6         y axis acceleration
+ *	data[2]        int32    micro g     -6              +/-4*10^6         z axis acceleration
+ *
+ * Ambient Light:
+ *	Data Fields    Type     Unit        Unit Exponent   Typical Range     Description
+ *	data[0]        uint32   milli lux   -3              [0, 10000] *10^3  Ambient light lux level
+ *
+ * The complete doc/spec should describe all supported sensors like above example.
+ */
+
+/**
+ * Sensor value data structure for quaternion.
+ * int16 version.
+ */
+struct senss_sensor_value_quaternion_int16 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
+		int16_t  x;
+		int16_t  y;
+		int16_t  z;
+		int16_t  w;
+	} readings[1];
+};
+
+/**
+ * Sensor value data structure for 3-aixs sensors.
+ * int32 version
+ */
+struct senss_sensor_value_3d_int32 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
+		union {
+			int32_t v[3];
+			struct {
+				int32_t x;
+				int32_t y;
+				int32_t z;
+			};
+		};
+	} readings[1];
+};
+
+/**
+ * Sensor value data structure for 2-aixs sensors.
+ * int32 version
+ */
+struct senss_sensor_value_2d_int32 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
+		union {
+			int32_t v[2];
+			struct {
+				int32_t x;
+				int32_t y;
+			};
+		};
+	} readings[1];
+};
+
+/**
+ * Sensor value data structure for single 1-aixs value.
+ * int32 version
+ */
+struct senss_sensor_value_int32 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
+		int32_t v;
+	} readings[1];
+};
+
+/**
+ * Sensor value data structure for single 1-aixs value.
+ * uint32 version
+ */
+struct senss_sensor_value_uint32 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
+		uint32_t v;
+	} readings[1];
+};
+
+/**
+ * Sensor value data structure for single 1-aixs value.
+ * int16 version
+ */
+struct senss_sensor_value_int16 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
+		int16_t v;
+	} readings[1];
+};
+
+/**
+ * Sensor value data structure for single 1-aixs value.
+ * uint16 version
+ */
+struct senss_sensor_value_uint16 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
+		uint16_t v;
+	} readings[1];
+};
+
+/**
+ * Sensor value data structure for single 1-aixs value.
+ * int8 version
+ */
+struct senss_sensor_value_int8 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
+		int8_t v;
+	} readings[1];
+};
+
+/**
+ * Sensor value data structure for single 1-aixs value.
+ * uint8 version
+ */
+struct senss_sensor_value_uint8 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
+		uint8_t v;
+	} readings[1];
+};
+
```

Comment:
Regarding to the data flows,  thanks for the suggestions,  yes,  a document with some diagrams will be very helpful,  will thinking about adding some documents.  Here are some simple descriptions:  data path is from bottom to up,  start from physical sensors to its client virtual sensors,  then to these virtual sensors' client virtual sensors ...,  go through whole DAG with data reporting path,  finally to the client Applications.  And config path is from up to bottom,  start from  client Applications to its accessed sensors,  then to these sensors' reporter sensors, finally to physical sensors and to sensor device drivers.  Senss will handle the arbitration both on data path and config path.  There are two data driven modes:  interrupt mode (triggered from sensor chip) and pooling mode (senss internal scheduling).  senss has a internal scheduler,  at current it's a single thread event process loop, its will handle these two kinds of data driven modes. 

Regarding to why need timestamp for each samples,  that's because in a not a very strict real time system, its hard to sampling sensor data with a accuracy fixed rate,  there are some small deviations,  such as some scheduling delay impacts.  But many sensor algorithms are sensitive on sensor data and its timestamp, such as integral operations in gyroscope,  so we still need record the timestamp for each sample when its generated. 

For sensor classes supported,  senss followed `HID` sensor types: 
https://github.com/zephyrproject-rtos/zephyr/pull/45370/commits/f4d2478e355b77b695588bb4d8bec9e22725c926#diff-21757224819cabfbdadaacbeeedf6c258c40e7124216bbe700586b7735967a65R14

And for new sensors such as lidars, radars, sonars,  we always keep strong interesting, and we have did some experiments on these sensors.